### PR TITLE
Some more documentation

### PR DESCRIPTION
**Pull request by jfalcou**

### CODE REVIEW

File: include/eve/literals.hpp [hunk 1]
```diff
@@ -29,27 +29,88 @@ namespace eve::detail
...
+  //! @groupheader{Wide literals}
+  //! Suffixes are provided to construct a eve::wide instance from a literal constant.
+  //!
+  //! @code
+  //! using namespace eve::literal;
+  //! // x is a eve::wide<short>
+  //! auto x = 45_wi16;
+  //!
+  //! // x is a eve::wide<float>
+  //! auto y = 2.635_wf;
```
Comment (DenisYaroshevskiy) [inline include/eve/literals.hpp:68]: Omg rly? :)
Why would we need this?

File: include/eve/platform.hpp [hunk 2]
```diff
@@ -12,28 +12,43 @@
...
+  //! @brief  Platform specific constexpr information
+  //!
+  //! Provides constexpr information about the status (supported or not) of various platform
+  //! computing capability.
+  //! @}
+  //================================================================================================
   struct platform
   {
+    //! @brief Evaluates to `true` if current platform supports NaN values
 #if defined(EVE_NO_NANS)
```
Comment (DenisYaroshevskiy) [inline include/eve/platform.hpp:28]: Is this rly a thing? Where?

File: include/eve/platform.hpp [hunk 3]
```diff
@@ -12,28 +12,43 @@
...
     static constexpr bool supports_nans = false;
 #else
     static constexpr bool supports_nans = true;
 #endif
 
+    //! @brief Evaluates to `true` if current platform supports infinites values
```
Comment (DenisYaroshevskiy) [inline include/eve/platform.hpp:34]: Same here? Who doesn't do IEEE?

**Review by DenisYaroshevskiy — APPROVED** [review]
I'm OK with this. I'd just remove most of it though.

File: include/eve/literals.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/literals.hpp:68]: It's an old piece of code I forgot existed and I documented it.
I am not attached to it so if you want we can remove it.

File: include/eve/platform.hpp (hunk 2 above)
Comment (jfalcou) [reply include/eve/platform.hpp:28]: A32 ARM doesn't support denormal values in SIMD register  for example. ARM code and tests use this supports_denormals cause if not, they fail.

The others are also meant to be passed by users in order to globally control the fact our SIMD functions care or do not care of some corner cases in FP to win some speed globally. 

I should add doc for the macro or if not this struct looks strange.

File: include/eve/platform.hpp (hunk 3 above)
Comment (jfalcou) [reply include/eve/platform.hpp:34]: See above

File: include/eve/literals.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/literals.hpp:68]: Let's remove them
